### PR TITLE
hide target category from cover overspending list

### PR DESCRIPTION
**Pull request by matt-fidd**
When using the cover overspending menu, it allows you to select the target category as the cover source. This doesn't cause a problem but doesn't make much sense or do anything.

### CODE REVIEW

File: packages/desktop-client/src/components/budget/rollover/CoverMenu.tsx [hunk 1]
```diff
@@ -1,20 +1,43 @@
...
 import { View } from '../../common/View';
 import { addToBeBudgetedGroup } from '../util';
 
+function removeSelectedCategory(
+  categoryGroups: CategoryGroupEntity[],
+  selectedCategoryId?: string,
+) {
+  if (!selectedCategoryId) return categoryGroups;
+
+  const newCategoryGroups: CategoryGroupEntity[] = JSON.parse(
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/rollover/CoverMenu.tsx:18]: Question: Why do we need to parse/stringify?

File: packages/desktop-client/src/components/budget/rollover/CoverMenu.tsx [hunk 2]
```diff
@@ -1,20 +1,43 @@
...
 type CoverMenuProps = {
   showToBeBudgeted?: boolean;
+  targetCategory?: string;
   onSubmit: (categoryId: string) => void;
   onClose: () => void;
 };
 
 export function CoverMenu({
   showToBeBudgeted = true,
+  targetCategory,
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/rollover/CoverMenu.tsx:40]: Nit:
```suggestion
  category,
```

**Review by joel-jeremy — CHANGES_REQUESTED** [review]

File: packages/desktop-client/src/components/budget/rollover/CoverMenu.tsx (hunk 1 above)
Comment (matt-fidd) [reply packages/desktop-client/src/components/budget/rollover/CoverMenu.tsx]: You're right, I was hitting issues with it changing the object in the higher scope as it was passed by ref. Reworked to use with `.map` now.

File: packages/desktop-client/src/components/budget/rollover/CoverMenu.tsx [hunk 3]
```diff
@@ -38,7 +56,7 @@ export function CoverMenu({
       <InitialFocus>
         {node => (
           <CategoryAutocomplete
-            categoryGroups={categoryGroups}
+            categoryGroups={removeSelectedCategory(categoryGroups, category)}
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/rollover/CoverMenu.tsx:59]: Would this be a good candidate for `useMemo`?

File: packages/desktop-client/src/components/budget/rollover/CoverMenu.tsx [hunk 4]
```diff
@@ -1,20 +1,38 @@
...
+    .map(group => ({
+      ...group,
+      categories: group.categories?.filter(cat => cat.id !== category),
+    }))
+    .filter(group => group.categories?.length);
+}
+
 type CoverMenuProps = {
   showToBeBudgeted?: boolean;
+  category?: string;
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/rollover/CoverMenu.tsx:28]:
```suggestion
  category?: CategoryEntity['id'];
```

File: packages/desktop-client/src/components/budget/rollover/CoverMenu.tsx (hunk 3 above)
Comment (matt-fidd) [reply packages/desktop-client/src/components/budget/rollover/CoverMenu.tsx]: I've added this now, it's a _relatively_ expensive calculation.

**Comment by joel-jeremy** [comment]
Thank you! Can we also apply this on the mobile category autocomplete modal?

I just also realized that it doesn't make sense to show categories with zero or negative balances on the cover category selection. But that's out of scope of this PR.

**Comment by matt-fidd** [comment]
@joel-jeremy where do I need to make the mobile changes please? I've dug about and I can't seem to find it, although I can see that on mobile it's different.

**Comment by joel-jeremy** [comment]
@matt-fidd It starts in the `CoverModal` which opens up the `CategoryAutocompleteModal`

**Comment by matt-fidd** [comment]
Phew that took a while!

@joel-jeremy - change now works on mobile too

**Comment by jfdoming** [comment]
Looks good! I'm thinking the same change would be good when performing a budget transfer (same menu option when a category is not overspent). Future PR?

**Comment by matt-fidd** [comment]
@jfdoming good shout! Now I know my way round this bit of the code base I'll handle that suggestion and earlier ones from joel-jeremy in a seperate PR.

**Review by jfdoming — APPROVED** [review]

**Pull request merged**